verticles/graphql/src/main/java/com/gentics/mesh/graphql/dataloader/DataLoaderKey.java

```diff
@@ -0,0 +1,53 @@
+package com.gentics.mesh.graphql.dataloader;
+
+import java.util.Objects;
+
+import org.dataloader.DataLoader;
+
+import graphql.execution.ResultPath;
+import graphql.schema.DataFetchingEnvironment;
+
+/**
+ * The GraphQL path-aware key for {@link DataLoader}.
+ * 
+ * @param <T>
+ */
+public final class DataLoaderKey<T> {
+
+	private final ResultPath path;
```

Comment:
So we take a GraphQL path into account, at last. So `data.rootMerged.contentChildren.elements` is separated from `data.rootMerged.folderChildren.elements`, and batch loader treat these two as separate queries, as the keys now differ.
Why this was working for non-native filtering? Because there were no filtering applied at the fetch stage, and so every query was ok with any data coming from the loader, performing the filtering afterwards.